test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs

```diff
@@ -77,431 +81,661 @@ public CpuMathUtilsUnitTests()
+
+            {"1", "-1.7", defaultMode},
+            {"1", "-1.7", disableAvx},
+            {"1", "-1.7", disableAvxAndSse},
+        };
+
+        public static TheoryData<string, string, string, string> MatMulData => new TheoryData<string, string, string, string>()
+        {
+            {"0", "0", "0", defaultMode},
+            {"1", "1", "0", defaultMode},
+            {"1", "0", "1", defaultMode},
+
+            {"0", "0", "0", disableAvx},
+            {"1", "1", "0", disableAvx},
+            {"1", "0", "1", disableAvx},
+
+            {"0", "0", "0", disableAvxAndSse},
+            {"1", "1", "0", disableAvxAndSse},
+            {"1", "0", "1", disableAvxAndSse},
+        };
+
         [Theory]
-        [InlineData(0, 0, 0, new float[] { -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f })]
-        [InlineData(1, 1, 0, new float[] { 1496f, 3672f, 5848f, 8024f, 10200f, 12376f, 14552f, 16728f })]
-        [InlineData(1, 0, 1, new float[] { 204f, 492f, 780f, 1068f, 1356f, 1644f, 1932f, 2220f, 2508f, 2796f, 3084f, 3372f, 3660f, 3948f, 4236f, 4524f })]
-        public void MatMulTest(int matTest, int srcTest, int dstTest, float[] expected)
+        [MemberData(nameof(MatMulData))]
+        public void MatMulTest(string matTest, string srcTest, string dstTest, string mode)
         {
-            AlignedArray mat = _testMatrices[matTest];
-            AlignedArray src = _testSrcVectors[srcTest];
-            AlignedArray dst = _testDstVectors[dstTest];
+            RemoteExecutorTestBase.RemoteInvoke((arg0, arg1, arg2, arg3) =>
+            {
+                CheckProperFlag(arg0);
+                AlignedArray mat = _testMatrices[Int32.Parse(arg1)];
+                AlignedArray src = _testSrcVectors[Int32.Parse(arg2)];
+                AlignedArray dst = _testDstVectors[Int32.Parse(arg3)];
+
+                float[] expected = new float[dst.Size];
+                for (int i = 0; i < dst.Size; i++)
+                {
+                    float dotProduct = 0;
+                    for (int j = 0; j < src.Size; j++)
+                    {
+                        dotProduct += mat[i * src.Size + j] * src[j];
+                    }                    
+                    expected[i] = dotProduct;
+                }
 
-            CpuMathUtils.MatrixTimesSource(false, mat, src, dst, dst.Size);
-            float[] actual = new float[dst.Size];
-            dst.CopyTo(actual, 0, dst.Size);
-            Assert.Equal(expected, actual, _matMulComparer);
+                CpuMathUtils.MatrixTimesSource(false, mat, src, dst, dst.Size);
+                float[] actual = new float[dst.Size];
+                dst.CopyTo(actual, 0, dst.Size);
+                Assert.Equal(expected, actual, _matMulComparer);
+                return RemoteExecutorTestBase.SuccessExitCode;
+            }, mode, matTest, srcTest, dstTest, new RemoteInvokeOptions() { mode = mode }).Dispose();
```

Comment:
Can `RemoteInvoke` just call `Dispose()`? That way every call site doesn't have to call it.